src/VirtoCommerce.XCart.Data/Services/CartProductService.cs

```diff
@@ -141,6 +141,23 @@ protected virtual async Task<List<CartProduct>> GetCartProductsAsync(IList<strin
             return result;
         }
 
+        protected virtual async Task<List<CartProduct>> LoadCartProductsAsync(CartProductsRequest request)
```

Comment:
LoadCartProductsASync does everything the same as GetCartProductsAsync. The function code is the same. Is it worth completely duplicating the code just because of the different input contract? Moreover, GetCartProductsAsync was used earlier, even in combination with CartProductsRequest. Wouldn't it be better to just extend GetCartProductsAsync with one additional EvaluatePromotions parameter? After all, the two functions do the same thing, but they have different names, which can be misleading.

Reply:
The idea is to eventually get rid of ICartProductService and switch to ICartProductsLoaderService